scripts/apps/knowledge-base/knowledge-base-page.tsx

```diff
@@ -0,0 +1,369 @@
+                                                            >
+                                                                {gettext('Remove')}
+                                                            </span>
+                                                        </button>
+                                                    </li>
+                                                </ul>
+                                            </Positioner>
+                                        </ListItemActionsMenu>
+                                    </ListItem>
+                                ))
+                            }
+                        </div>
+                    </PageContainerItem>
+
+                    {
+                        this.state.itemInPreview != null ? (
+                            <PageContainerItem>
+                                <KnowledgeItemViewEdit
+                                    onEditModeChange={(val) => {
+                                        this.previewInEditMode = val;
+                                    }}
+                                    operation="editing"
+                                    formConfig={formConfig}
+                                    item={
+                                        this.props.conceptItems._items.find(({_id}) => _id === this.state.itemInPreview)
+                                    }
+                                    onSave={(nextItem) => this.props.conceptItems.update(nextItem)}
+                                    onClose={this.closePreview}
+                                />
+                            </PageContainerItem>
+                        ) : null
+                    }
+
+                    {
+                        this.state.newItem != null ? (
+                            <PageContainerItem>
+                                <KnowledgeItemViewEdit
+                                    operation="creation"
+                                    formConfig={formConfig}
+                                    item={this.state.newItem}
+                                    onSave={(item: IKnowledgeBaseItem) => this.props.conceptItems.create({
+                                        ...item,
+                                        cpnat_type: 'cpnat:abstract',
+                                    }).then((res) => {
+                                        this.closeNewItemForm();
+                                        this.openPreview(res._id);
+                                    })}
+                                    onClose={this.closeNewItemForm}
+                                    onCancel={this.closeNewItemForm}
+                                />
+                            </PageContainerItem>
+                        ) : null
+                    }
+                </PageContainer>
+            </div>
+        );
+    }
+}
+
+export const KnowledgeBasePage = connectServices(
```

Comment:
can you write some tests for this component and connectCrudManager ?

Reply:
`CrudManager` is pretty straight-forward and doesn't do much but API requests, so not easy to test it by mocking nor writing e2e tests since it's a standalone component.

I'll see what I can come up with for the page itself, I guess I'll write a few e2e tests.

Reply:
I've removed _knowledge-base_ related specifics and created a `GenericListPageComponent`. I refactored _internal destinations settings page_ to use the component and have written tests for it at `spec/internal_destinations_spec.js`